src/test/java/uk/gov/companieshouse/itemhandler/kafka/ItemMessageProducerIntegrationTest.java

```diff
@@ -0,0 +1,111 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.test.annotation.DirtiesContext;
+import uk.gov.companieshouse.itemhandler.model.Item;
+import uk.gov.companieshouse.kafka.message.Message;
+import uk.gov.companieshouse.kafka.serialization.AvroSerializer;
+import uk.gov.companieshouse.kafka.serialization.SerializerFactory;
+import uk.gov.companieshouse.orders.OrderReceived;
+
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.springframework.context.annotation.FilterType.ASSIGNABLE_TYPE;
+import static uk.gov.companieshouse.itemhandler.util.TestConstants.MISSING_IMAGE_DELIVERY_ITEM_ID;
+import static uk.gov.companieshouse.itemhandler.util.TestConstants.ORDER_REFERENCE;
+
+@SpringBootTest
+@DirtiesContext
+@EmbeddedKafka
+class ItemMessageProducerIntegrationTest {
+
+    @Configuration
+    @ComponentScan(basePackageClasses = ItemMessageProducerIntegrationTest.class,
+                   excludeFilters = {@ComponentScan.Filter(type = ASSIGNABLE_TYPE,
+                                                           value = OrdersKafkaConsumerWrapper.class),
+                                     @ComponentScan.Filter(type = ASSIGNABLE_TYPE,
+                                                           value = OrdersKafkaConsumer.class)})
+    public static class Config {
+        @Bean
+        public SerializerFactory getSerializerFactory() {
+            return new SerializerFactory();
+        }
+
+        @Bean
+        public ItemMessageProducer itemMessageProducerUnderTest() {
+            return new ItemMessageProducer(getItemMessageFactory(), getItemKafkaProducer());
+        }
+
+        @Bean
+        public ItemMessageFactory getItemMessageFactory() {
+            return new ItemMessageFactory(getSerializerFactory());
+        }
+
+        @Bean
+        public ItemKafkaProducer getItemKafkaProducer() {
+            return new ItemKafkaProducer();
+        }
+    }
+
+    @Autowired
+    private ItemMessageProducer itemMessageProducerUnderTest;
+
+    @Autowired
+    private TestItemMessageConsumer testItemMessageConsumer;
+
+    @Autowired
+    private SerializerFactory serializerFactory;
+
+    @Test
+    void testSendItemMessageToKafkaTopic() throws Exception {
+
+        // Given an item is to be sent
+        final Item item = new Item();
+        item.setId(MISSING_IMAGE_DELIVERY_ITEM_ID);
+
+        // Given that for now the actual message produced to the topic is an OrderReceived object.
+        // TODO GCI-1301 Replace hijacked OrderReceived kafka-models class with a class for chd-item-ordered.
```

Comment:
I wonder if these `TODO`s can be removed and a link to this line be added to the Jira story that will handle this as developer note. It will keep Sonar happy.

Reply:
That's a good idea. I will do that. 👍